tools/webdriver/webdriver/client.py

```diff
@@ -752,6 +758,60 @@ def execute_async_script(self, script, args=None):
     def screenshot(self):
         return self.send_session_command("GET", "screenshot")
 
+class BidiSession(Session):
+    def __init__(self,
+                 host,
+                 port,
+                 url_prefix="/",
+                 capabilities=None,
+                 extension=None):
+        """
+        Add a capability of "webSocketUrl": True to enable
+        Bidirectional connection in session creation.
+        """
+        self.websocket_transport = None
+        capabilities = self.__enable_websocket(capabilities)
+        super().__init__(host, port, url_prefix, capabilities, extension)
+
+    def __enable_websocket(self, caps):
+        if caps:
+            caps.setdefault("alwaysMatch", {}).update({"webSocketUrl": True})
+        else:
+            caps = {"alwaysMatch": {"webSocketUrl": True}}
+        return caps
+
+    def match(self, capabilities):
+        """Expensive match to see if capabilities is the same as previously
+        requested capabilities if websocket would be enabled.
+
+        :return Boolean.
+        """
+        caps = copy.deepcopy(capabilities)
+        caps = self.__enable_websocket(caps)
+        return super().match(caps)
+
+    async def start(self):
+        """Start a new WebDriver Bidirectional session
+        with websocket connected.
+
+        :return: Dictionary with `capabilities` and `sessionId`.
+        """
+        value = super().start()
+        # Refresh websocket connection for every test to have a clean
```

Comment:
I don't think this library should contain this kind of test-specific detail, but it should go into `conftest.py`. Closing the websocket connection on every test seems pretty dubious; there isn't much state tied to the connection itself rather than the session, so I'd initially not do this and see if we are running into issues.

Reply:
Removed the close and re-connect logic.